ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,314 @@
+            $self->error("Cannot handle $2 (non-1K) AFS cache block sizes");
+            return 0;
+        }
+        $run_cache = $1;
+    }
+    else {
+        $self->info("Cannot determine current AFS cache size, changing only config file");
+    }
+
+    my $afs_cacheinfo_fh = CAF::FileEditor->new( $afs_cacheinfo, log => $self );
+    if ( "$afs_cacheinfo_fh" =~ m;^([^:]+):([^:]+):(\d+)$; ) {
+        $file_afsmount   = $1;
+        $file_cachemount = $2;
+        $file_cache      = $3;
+    }
+    else {
+        $self->error("Cannot parse stored AFS cache mount or size from $afs_cacheinfo");
+        return 0;
+    }
+    $afs_cacheinfo_fh->close();
+
+    # sanity check - don't allow cachesize bigger than 95% of a partition size
+    $proc = CAF::Process->new( [ "df", "-k", $file_cachemount ], log => $self, keeps_state => 1 );
+    for ( split ( "\n", $proc->output() ) ) {
+        if (m{^(.*?)\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+%)\s+(.*)}) {
+            my $disk_cachesize = $2;
+            my $mount          = $6;
+            if ( $mount eq $file_cachemount && $new_cache > 0.95 * $disk_cachesize ) {
+                $self->error("Cache size cannot exceed 95% of its partition size. Not changing.");
+                return 0;
+            }
+        }
     }
 
-    $ret &= NCM::Check::lines($sysconfig_afs,
-			     linere => '^#.*$compname-enabled|AFS_CLIENT=.*',
-			     goodre => '^#.*$compname-enabled|AFS_CLIENT=\s*'.quotemeta($enabled),
-			     good   => "# changed by $compname-enabled\nAFS_CLIENT=".$enabled,
-			     keep   => 'first',
-			     add    => 'last',
-			     backup => '.old_enabled');
-  }
-
-  if($config->elementExists($mypath."/afsmount")) {
-    my $new_afsmount = $config->getValue($mypath."/afsmount");
-    if ($new_afsmount ne '/afs') {
-      $self->error("cannot handle non-/afs AFS mount point $new_afsmount");
-      return 0;
+    # adjust stored cache size (gets overwritten on restart for OpenAFS-1.4)
+    if ( $new_cache != $file_cache ) {
+        my $afs_cacheinfo_fh = CAF::FileWriter->new( $afs_cacheinfo, log => $self );
+        print $afs_cacheinfo_fh "$file_afsmount:$file_cachemount:$new_cache\n";
+        if ( $afs_cacheinfo_fh->close() ) {
+            $self->info("Changed AFS cache config file $afs_cacheinfo: $file_cachemount $run_cache -> $new_cache (1K blocks)");
+        }
     }
-  }
 
-  return $ret;
+    # adjust online (in-kernel) value
+    if ( $run_cache && ( $new_cache ne "AUTOMATIC" ) && ( $run_cache != $new_cache ) ) {
+        if ($NoAction) {
+            $self->info("Would need to update running cache size");
+        }
+        else {
+            my $fscommand = [ "fs", "setcachesize", $new_cache ];
+            $proc = CAF::Process->new( $fscommand, log => $self );
+            $proc->run();
+            if ( $? >> 8 ) {
+                $self->warn( "Problem changing running AFS cache size via \"" . join ( " ", @$fscommand ) . "\":\n" + $proc->output() );
```

Comment:
`CAF::Process` has stringification, so instead of `join ( " ", @$fscommand )`, you can just `"$proc"`
`proc->output()` will run the command a second time; so better to `$output = $proc->output()` instead of `->run` (i also think `run` dumps output to stdout, which noone wants)